app/src/main/kotlin/io/homeassistant/companion/android/widgets/BaseGlanceEntityWidgetReceiver.kt

```diff
@@ -0,0 +1,289 @@
+        val appWidgetId = intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, AppWidgetManager.INVALID_APPWIDGET_ID)
+        Timber.v("Received intent action = ${intent.action} for widget = $appWidgetId ($widgetClassName) $this")
+
+        // Handle the Glance [androidx.glance.action.Action]
+        super.onReceive(context, intent)
+
+        when (intent.action) {
+            Intent.ACTION_SCREEN_ON -> watchingForEntitiesChanges(context)
+            Intent.ACTION_SCREEN_OFF -> stopWatchingForEntitiesChanges()
+            AppWidgetManager.ACTION_APPWIDGET_UPDATE -> {
+                /*
+                 * WARNING: This action is received from another object thant the other above so you shouldn't entities, otherwise it
+                 * won't ever be cancel.
+                 *
+                 * This will trigger an update of all the widget, but won't watch for changes.
+                 *
+                 * <p>This may be sent in response to a new instance for this AppWidget provider having
+                 * been instantiated, the requested {@link AppWidgetProviderInfo#updatePeriodMillis update interval}
+                 * having lapsed, or the system booting.
+                 */
+                widgetScope.launch {
+                    Timber.d("Update all widgets")
+                    glanceAppWidget.updateAll(context)
+                }
+            }
+        }
+    }
+
+    override fun onDeleted(context: Context, appWidgetIds: IntArray) {
+        Timber.d("onDeleted $appWidgetIds")
+        super.onDeleted(context, appWidgetIds)
+        deleteWidgetsFromDatabase(appWidgetIds)
+    }
+
+    /**
+     * Register this receiver to receive [Intent.ACTION_SCREEN_ON] and [Intent.ACTION_SCREEN_OFF].
+     * It doesn't exported the receiver.
+     */
+    fun registerReceiver(context: Context) {
+        Timber.tag(widgetClassName).d("Register receiver ($this)")
+        ContextCompat.registerReceiver(
+            context,
+            this,
+            IntentFilter().apply {
+                addAction(Intent.ACTION_SCREEN_OFF)
+                addAction(Intent.ACTION_SCREEN_ON)
+            },
+            ContextCompat.RECEIVER_NOT_EXPORTED,
+        )
+    }
+
+    private fun deleteWidgetsFromDatabase(appWidgetIds: IntArray) {
+        widgetScope.launch {
+            Timber.tag(widgetClassName).d("onDeleted ${appWidgetIds.toList()}")
+            dao.deleteAll(appWidgetIds)
+            appWidgetIds.forEach(::removeSubscription)
+        }
+    }
+
+    private fun setupWidgetScope() {
+        if (!widgetScope.isActive) {
+            widgetScope = widgetScopeProvider()
+        }
+    }
+
+    private fun watchingForEntitiesChanges(context: Context) {
+        setupWidgetScope()
+        if (!serverManager.isRegistered()) {
+            Timber.tag(widgetClassName).d("No server registered won't watch for entities")
+            return
+        }
+
+        Timber.e("Watching for entities $widgetScope")
```

Comment:
Why is this an error? Did you leave in debug logging statements for development and if so do you plan to clean them up?

Reply:
Mistake, I did clean up but I had so many debugs logs that I might have missed some like this one.